build-logic-settings/testfiltering-plugin/src/main/kotlin/gradlebuild.internal.testfiltering.settings.gradle.kts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+val testFilteringProperty = "gradle.internal.testselection.enabled"
+val gitBranchName: String? = System.getenv("BUILD_BRANCH")
+if (!System.getProperties().containsKey(testFilteringProperty) && gitBranchName != null) {
```

Comment:
The intent here is to do nothing if the property is already set (whether true or false), in order to prevent this code from overriding a preset value.

Reply:
Thanks for explanation!